lib/at_shell/at_shell.c

```diff
@@ -54,6 +54,18 @@ int at_shell(const struct shell *shell, size_t argc, char **argv)
 		at_monitor_pause(&at_shell_monitor);
 		shell_print(shell, "AT command event handler disabled");
 	} else {
+		if (IS_ENABLED(CONFIG_AT_SHELL_UNESCAPE_LF)) {
+			char *c = command;
+
+			/* Replace \n with <CR><LF> so commands which accept multiline input
+			 * such as AT%CMNG=0 will work with a properly pre-processed cert.
+			 */
+			while ((c = strstr(c, "\\n")) != NULL) {
+				c[0] = '\r';
+				c[1] = '\n';
+			}
```

Comment:
IMO, if we add escapement for `\n` we also need to add escapement for `\`. Otherwise we are replacing one set of impossible-to-encode strings with another

Reply:
This does function properly as I wrote it. I do not see an issue.

Reply:
In your use case, since the payload is base-64 it will not contain any backslash-n's, sure

But there may, at some point (or perhaps may already be) an AT command where an input such as "s\fhn\snjf\hs\nad\fnjh\sd\f" is valid. 

Admittedly, I can't think of any, so maybe it doesn't exist. But to me it seems better to support `\\` escapes, so that any users of this feature don't have to question whether the input they provide is going to contain `\n` incidentally

Reply:
What this does is replace the two characters backslash (\) and en (n) with the two characters 0x0D 0x0A (<CR><LF) in the string being sent to the modem. I don't understand your concern.

I am changing the code so that it is only used for `AT%CMNG=` based on Justin's comment.

Reply:
> What this does is replace the two characters backslash () and en (n) with the two characters 0x0D 0x0A (<LF) in the string being sent to the modem. I don't understand your concern.

I understand. If you are restricting this to `AT%CMNG`, then it doesn't matter to me anymore, because my concern becomes verifiably impossible.

But to clarify, my concern was that doing so makes it impossible to send `\n` -- literally, backslash + n, not newline, in any quoted AT arguments